src/celeritas/optical/surface/model/SurfaceInteraction.hh

```diff
@@ -0,0 +1,77 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/optical/surface/model/SurfaceInteraction.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/math/ArrayUtils.hh"
+#include "corecel/math/SoftEqual.hh"
+
+namespace celeritas
+{
+namespace optical
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Representation of a photon's direction and polarization as a phasor.
+ */
+struct PhotonPhasor
+{
+    Real3 direction;
+    Real3 polarization;
+
+    //! Whether the phaser is a valid photon state
+    explicit CELER_FUNCTION operator bool() const
```

Comment:
This operation is expensive enough that I think it should be called something else (it should be used only for error checking, not for program logic).